src/backend/src/services/projects.services.ts

```diff
@@ -341,15 +329,57 @@ export default class ProjectsService {
       });
     }
 
-    addDescriptionBullets(goalsChangeJson.addedDetails, updatedProject.projectId, 'projectIdGoals');
-    addDescriptionBullets(featuresChangeJson.addedDetails, updatedProject.projectId, 'projectIdFeatures');
-    addDescriptionBullets(otherConstraintsChangeJson.addedDetails, updatedProject.projectId, 'projectIdOtherConstraints');
-    editDescriptionBullets(
-      goalsChangeJson.editedIdsAndDetails
-        .concat(featuresChangeJson.editedIdsAndDetails)
-        .concat(otherConstraintsChangeJson.editedIdsAndDetails)
+    await addDescriptionBullets(
+      goalsChangeJson.addedElements.map((descriptionBullet) => descriptionBullet.detail),
+      updatedProject.projectId,
+      'projectIdGoals'
+    );
+    await addDescriptionBullets(
+      featuresChangeJson.addedElements.map((descriptionBullet) => descriptionBullet.detail),
+      updatedProject.projectId,
+      'projectIdFeatures'
+    );
+    await addDescriptionBullets(
+      otherConstraintsChangeJson.addedElements.map((descriptionBullet) => descriptionBullet.detail),
+      updatedProject.projectId,
+      'projectIdOtherConstraints'
+    );
+    await editDescriptionBullets(
+      goalsChangeJson.editedElements
+        .concat(featuresChangeJson.editedElements)
+        .concat(otherConstraintsChangeJson.editedElements)
     );
 
+    await linkChanges.addedElements.forEach(async (link) => {
```

Comment:
do we want to make these three things its own function

Reply:
yeah that would make it a little cleaner